### ISSUE TITLE
hasManyDeepFromRelations Does not use with the 'wheres' of the first

### ISSUE THREAD

**Issue by sbc640964** [staudenmeir/eloquent-has-many-deep#161]
I have two functions that return relationship - 1:records 2:data
For the first time I added wheres inside the function
```php
public function records()
{
    return $this->hasMenyDeep(...).where(...);
}

public function data()
{
    return $this->hasMany(Data::class);
}
```
When I use with hasManyDeepFromRelations I get a query that does not include the queries of the records function:
```php
return hasManyDeepFromRelations(
    $this->records(),
    (new Record())->data()
);
```

**Comment by staudenmeir** [comment]
`hasManyDeepFromRelations()` does not yet apply constraints, you need to add them manually for now:

```php
return hasManyDeepFromRelations(
    $this->records(),
    (new Record())->data()
)->where(...);
```

**Comment by sbc640964** [comment]
@staudenmeir  Thanks.
Too bad, why not really?

**Comment by staudenmeir** [comment]
It's quite complex to cover all the possible cases, but I'm working on it.

**Comment by staudenmeir** [comment]
I released `v1.15.3` with [`hasManyDeepFromRelationsWithConstraints()`](https://github.com/staudenmeir/eloquent-has-many-deep#existing-relationships-constraints):

```php
return $this->hasManyDeepFromRelationsWithConstraints(
    [$this, 'records'],
    [new Record(), 'data']
);
```
